### PR TITLE
feat(frontend): Reorganize Experiments, Runs, and Archive pages. Fixes #4918

### LINKED ISSUE #4918: Reorganize Experiments, Runs, and Archive pages
Currently KFP has the following sidebar items:

- Pipelines
- Experiments
    - Tab1: Experiments
    - Tab2: Runs
- Archive
	- Tab1: Archived Experiments
	- Tab2: Archived Runs
- Artifacts
- Executions

original-exps
original-archive

KFP effectively deals with three main "entities" (+ MLMD ones), that the user creates and interacts with:

1. Pipelines
2. Experiments
3. Runs

What we (Arrikto) have realized after taking a lot of feedfback from our users, is that the current layout makes navigating these entities somewhat confusing, and it doesn't promote Runs equally as they deserve. We think that the following slight re-organization improves significantly the overall UX and helps less experienced users get onboarded much quicker.

Our revised layout reorganizes the current pages as follows:

- Pipelines
- Experiments
	- Tab1: Active
	- Tab2: Archived
- Runs
	- Tab1: Active
	- Tab2: Archived
- Artifacts
- Executions

new-exps-active
new-runs-active

Note that this does not imply **any** functional change, just a rearrangement of the existing pages.

Additionally, we have also developed a new page to track and monitor recurring runs that we plan to contribute upstream very soon (you can see the sidebar entry in the screenshot above), and with the proposed new layout every top-level entity is exposed uniformly to the user.

I will follow up with a PR with the proposed implementation, and we can keep further discussion here.

### PR DESCRIPTION
**Pull request by StefanoFioravanzo**
**Description of your changes:**

Improve the semantic organization of the KFP pages by reorganizing Experiments, Runs and Archive pages as follows:

- Pipelines
- Experiments
    - Tab1: Active
    - Tab2: Archived
- Runs
    - Tab1: Active
    - Tab2: Archived
- Artifacts
- Executions
- ~~Archive~~
 
More details at #4918

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by StefanoFioravanzo** [comment]
I am not sure why the e2e tests are failing. @Bobgy @Ark-kun maybe you have more insight?

**Comment by Bobgy** [comment]
FYI, frontend-integration test (selenium based) is failing because the reorganization affected assumptions in the test.
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/4919/kubeflow-pipeline-e2e-test/1339901843387453440#1:build-log.txt%3A5471

Test code at https://github.com/kubeflow/pipelines/blob/master/test/frontend-integration-test/helloworld.spec.js

**Comment by Bobgy** [comment]
I'll take another look when you fixed integration test.

The idea sounds great to me!
Thank you so much for the contribution!

**Comment by StefanoFioravanzo** [comment]
@Bobgy Thanks for the tip! I fixed the frontend integration test

**Comment by StefanoFioravanzo** [comment]
Hey @Bobgy do you think we could merge this before Christmas? :)

File: frontend/src/components/Router.tsx [hunk 1]
```diff
@@ -159,23 +159,23 @@ const Router: React.FC<RouterProps> = ({ configs }) => {
   const routes: RouteConfig[] = configs || [
     { path: RoutePage.START, Component: GettingStarted },
     {
-      Component: ArchivedExperimentsAndRuns,
+      Component: AllRunsAndArchive,
       path: RoutePage.ARCHIVED_RUNS,
```
Comment (Bobgy) [inline frontend/src/components/Router.tsx:163]: I think in the new structure, it makes more sense to change the route from `/archive/runs` to `/runs/archive`, what do you think? Maybe we can do it in a separate PR, because route change means we might need to set up redirect from the old route to the new route to ease the migration if people have built systems that take KFP UI URL.

File: frontend/src/pages/AllExperimentsAndArchive.test.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 Google LLC
+ * Copyright 2018 Google LLC
```
Comment (Bobgy) [inline frontend/src/pages/AllExperimentsAndArchive.test.tsx:2]: nit: 2020?

**Comment by Bobgy** [comment]
Thanks @StefanoFioravanzo!
I think the code change is great!
/lgtm
/approve

**Review by Bobgy — COMMENTED** [review]
Some minor comments only

**Comment by Bobgy** [comment]
/unhold
I think we can skip the minor comments for now

File: frontend/src/components/Router.tsx (hunk 1 above)
Comment (StefanoFioravanzo) [reply frontend/src/components/Router.tsx:163]: I definitely agree on this and actually wanted to propose it as a future improvement. I'll think about it and propose next steps
